tests/test_binding_inheritance.py

```diff
@@ -0,0 +1,610 @@
+        pilot.app.all_recorded("screenly_")
+
+
+##############################################################################
+# Testing priority of overlapping bindings.
+#
+# Here we we'll have an app, screen, and a focused widget, along with a
+# combination of overlapping bindings, each with different forms of
+# priority, so we can check who wins where.
+#
+# Here are the permurations tested, with the expected winner:
+#
+# |-----|----------|----------|----------|--------|
+# | Key | App      | Screen   | Widget   | Winner |
+# |-----|----------|----------|----------|--------|
+# | 0   |          |          |          | Widget |
+# | A   | Priority |          |          | App    |
+# | B   |          | Priority |          | Screen |
+# | C   |          |          | Priority | Widget |
+# | D   | Priority | Priority |          | App    |
+# | E   | Priority |          | Priority | App    |
+# | F   |          | Priority | Priority | Screen |
+
+
+class PriorityOverlapWidget(Static, can_focus=True):
+    """A focusable widget with a priority binding."""
+
+    BINDINGS = [
+        Binding("0", "app.record('widget_0')", "0", priority=False),
+        Binding("a", "app.record('widget_a')", "a", priority=False),
+        Binding("b", "app.record('widget_b')", "b", priority=False),
+        Binding("c", "app.record('widget_c')", "c", priority=True),
+        Binding("d", "app.record('widget_d')", "d", priority=False),
+        Binding("e", "app.record('widget_e')", "e", priority=True),
+        Binding("f", "app.record('widget_f')", "f", priority=True),
+    ]
+
+
+class PriorityOverlapScreen(Screen):
+    """A screen with a priority binding."""
+
+    BINDINGS = [
+        Binding("0", "app.record('screen_0')", "0", priority=False),
+        Binding("a", "app.record('screen_a')", "a", priority=False),
+        Binding("b", "app.record('screen_b')", "b", priority=True),
+        Binding("c", "app.record('screen_c')", "c", priority=False),
+        Binding("d", "app.record('screen_d')", "c", priority=True),
+        Binding("e", "app.record('screen_e')", "e", priority=False),
+        Binding("f", "app.record('screen_f')", "f", priority=True),
+    ]
+
+    def compose(self) -> ComposeResult:
+        yield PriorityOverlapWidget()
+
+    def on_mount(self) -> None:
+        self.query_one(PriorityOverlapWidget).focus()
+
+
+class PriorityOverlapApp(AppKeyRecorder):
+    """An application with a priority binding."""
+
+    BINDINGS = [
+        Binding("0", "record('app_0')", "0", priority=False),
+        Binding("a", "record('app_a')", "a", priority=True),
+        Binding("b", "record('app_b')", "b", priority=False),
+        Binding("c", "record('app_c')", "c", priority=False),
+        Binding("d", "record('app_d')", "c", priority=True),
+        Binding("e", "record('app_e')", "e", priority=True),
+        Binding("f", "record('app_f')", "f", priority=False),
+    ]
+
+    SCREENS = {"main": PriorityOverlapScreen()}
```

Comment:
This may be the source of your test fail.

Reply:
Interesting, I'll give that a try. I *think* I turned it into an instance at some point as a process of elimination but forgot to swap back.

Any idea as to why this would cause a problem, especially on different Python versions? (the issues I'm seeing, so far, seem to be down to Python version).

Reply:
Not entirely sure. I think it is something to do with creating async primitives in different loops, as created by pytest.

Reply:
FWIW, that did fix it.